src/charm.py

```diff
@@ -0,0 +1,163 @@
+        )
+        self.framework.observe(self.on.vault_pebble_ready, self._on_config_changed)
+        self.framework.observe(self.on.new_policy_action, self._new_policy_action)
+        self.framework.observe(self.on.new_app_role_action, self._new_app_role_action)
+        self.framework.observe(self.on.get_token_action, self._get_token_action)
+        self.framework.observe(self.on.get_root_token_action, self._get_root_token_action)
+        self.framework.observe(self.peers.on.has_peers, self._on_config_changed)
+
+    def _on_certificate_request(self, event):
+        certificate = self.vault.issue_certificate(
+            cert_type=event.cert_type, common_name=event.common_name, sans=event.sans
+        )
+        self.tls_certificates.set_relation_certificate(
+            certificate=Cert(
+                cert=certificate["certificate"],
+                key=certificate["private_key"],
+                ca=certificate["issuing_ca"],
+                common_name=event.common_name,
+            ),
+            relation_id=event.relation_id,
+        )
+
+    def _on_config_changed(self, event):
+        """Handle the config-changed event."""
+        container = self.unit.get_container(self._container_name)
+        if not container.can_connect():
+            event.defer()
+            return
+        plan = container.get_plan()
+        layer = self._vault_layer()
+        if plan.services != layer.services:
+            container.add_layer(self._container_name, layer, combine=True)
+            logging.info("Added updated layer 'vault' to Pebble plan")
+            container.restart(self._service_name)
+            logging.info("Restarted vault service")
+        # Fix up storage permissions (broken on CDK on AWS otherwise)'
+        os.chown(STORAGE_PATH, uid=100, gid=1000)
+
+        if self.unit.is_leader():
+            if not self.vault.is_available:
+                logger.info("Vault is not available, deferring")
+                event.defer()
+                return
+            if not self.vault.is_initialized:
+                self.vault.initialize()
+                self.peers.set_root_token(self.vault.root_token)  # type: ignore[arg-type]
+                self.peers.set_unseal_key(self.vault.unseal_key)  # type: ignore[arg-type]
+            if self.vault.is_sealed:
+                self.vault.unseal()
+            root_ca = self.vault.generate_root_ca()
+            self.peers.set_root_ca(root_ca)
+        self.unit.status = ActiveStatus()
+
+    def get_ca(self) -> str:
+        """Returns the root_ca."""
+        return self.peers.root_ca
+
+    @property
+    def _bind_address(self) -> str:
+        peer_relation = self.model.get_relation("peers")
+        return str(self.model.get_binding(peer_relation).network.bind_address)  # type: ignore[arg-type, union-attr]  # noqa: E501
+
+    def _vault_layer(self) -> Layer:
+        backends = {"file": {"path": "/srv"}}
```

Comment:
Here you're using `/srv` and not the storage path you define earlier :)

Reply:
Let's use the thread above to resolve this.